db-migration/run.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/sh
+
+echo 'Logging into Gov PaaS'
+
+if cf login -a api.london.cloud.service.gov.uk -u ${CF_USERNAME} -p ${CF_PASSWORD} -s ${CF_SPACE}; then
+  echo 'Logged into Gov PaaS successfully'
+else
+  echo 'ERROR could not log into Gov Paas'
+  exit 1
+fi
+
+echo "Setting up conduit into Gov PaaS DB and making dump file"
+
+if cf conduit ${CF_SERVICE} -- pg_dump --file dumpfile --no-acl --encoding utf8 --clean --no-owner --if-exists -Fc; then
+  echo 'Dump file of PaaS DB created successfully'
+else
+  echo 'ERROR trying to conduit into the PaaS DB or creating the dump file'
+  exit 1
+fi
+
+while [ ! -e  "$dumpfile"]; do
+  echo "Waiting for dump file to be ready for pg restore"
+  sleep 1
+done
+
+echo 'Dump file ready for pg restore'
+echo 'Starting pg restore of AWS DB'
+
+if pg_restore -v -d ${DATABASE_URL} -j 4 --no-acl --clean --no-owner --if-exists -Fc dumpfile; then
+  echo 'Ran pg restore on AWS DB successfully'
+  exit 0
+else
+  echo 'ERROR running pg restore on AWS DB. Check if the errors reported are important or harmless as the restore may still have completed ok'
+  exit 1
+fi
```

Comment:
Question: if an error does happen during this process, what can we do to deal with it? (It may be that our best option is to start again if anything important has failed, but we should make sure we know that)

Reply:
Good question! My thinking here is that all the steps are pretty critical so if one fails, we'll probably want to investigate and start over. I think it is possible to run the` pg_restore` over a DB (believe it is idempotent although it might report errors about things already existing), however depending on the kind of error we may want to clean out the DB and start from green again.

Also we have the exec placeholder task in order to be able to re-run the commands and try and diagnose in more detail what's going on.

Reply:
Documentation updates done 👍